frontend/lib/framework/error-classes-list.cpp

```diff
@@ -720,6 +721,46 @@ void ErrorBisonSyntaxError::write(ErrorWriterBase& wr) const {
   wr.code(loc);
 }
 
+/* post-parse-checks errors */
+void ErrorPostParseErr::write(ErrorWriterBase& wr) const {
+  auto id = std::get<const ID>(info);
+  auto errorMessage = std::get<std::string>(info);
+  CHPL_ASSERT(errorMessage.back() == '.' &&
+         "expected a period at the end of ErrorPostParseErr message");
+  wr.heading(kind_, type_, id, errorMessage);
+  wr.code(id);
+}
+
+void ErrorPostParseWarn::write(ErrorWriterBase& wr) const {
+  auto id = std::get<const ID>(info);
+  auto errorMessage = std::get<std::string>(info);
+  CHPL_ASSERT(errorMessage.back() == '.' &&
+         "expected a period at the end of ErrorPostParseWarn message");
+  wr.heading(kind_, type_, id, errorMessage);
+  wr.code(id);
+}
+
+void ErrorMultipleClassKinds::write(ErrorWriterBase& wr) const {
+  auto id = std::get<const ID>(info);
+  auto outerMgt = std::get<1>(info);
+  auto innerMgt = std::get<2>(info);
+  wr.heading(kind_, type_, id,
+             "type expression uses multiple class kinds: ", outerMgt, " ",
+             innerMgt, ".");
+  wr.message("Multiple class kinds used in type expression here:");
+  wr.code(id);
+  wr.message("Only a single management kind can be used in a type expression.");
```

Comment:
We have the opportunity to say "remove either `innerMgt` or `outerMgt`", or better yet, if there are two:
1. if they are the same, say one of them is redundant
2. if they are different, say they're incompatible and to remove one.